blue_bringup/scripts/gripper_calibration_service.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+
+"""This node should run at startup, and sets the initial joint angles to some hardcoded values."""
+
+import rospy
+import numpy as np
+import actionlib
+from blue_msgs.srv import JointStartupCalibration
+from controller_manager_msgs.srv import SwitchController
+from controller_manager_msgs.srv import LoadController
+from std_srvs.srv import Trigger, TriggerRequest, TriggerResponse
+from blue_msgs.msg import MotorState
+from std_msgs.msg import Float64MultiArray
+from control_msgs.msg import (
+    GripperCommandAction,
+    GripperCommandGoal,
+)
+
+def update_motors(motor_msg):
+    global gripper_state
+    global motor_names
+    global disable_snap
+    global startup_positions
+    for i, n in enumerate(motor_msg.name):
+        if n == motor_names[-1]:
+            gripper_state = motor_msg.position[i]
+            break
+
+def handle_calibration_service(input_request):
+    global gripper_state
+    global motor_names
+    global disable_snap
+    global startup_positions
+
+    global joint_startup_calibration
+    global switch_controller
+    global cmd_pub
+
+    # Calibrate joints with startup angles
+    rospy.loginfo("Starting calibration...")
+    try:
+        gripper_controller_response = switch_controller(['blue_controllers/gripper_torque_controller'], [], 2)
+
+        cmd = Float64MultiArray()
+        cmd.data = [5.0]
+        # apply positive torque
+        for i in range(5):
+            cmd_pub.publish(cmd)
+            rate.sleep()
+        rospy.wait_for_message("blue_hardware/motor_states", MotorState)
+
+        current_position = gripper_state + 10
+        # busy wait until gripper is closed
+        while not rospy.is_shutdown() and np.abs(current_position - gripper_state) > 1e-5:
+            # apply positive torque
+            # gripper still closing
+            cmd_pub.publish(cmd)
+            current_position = gripper_state
+            rospy.wait_for_message("blue_hardware/motor_states", MotorState)
```

Comment:
this returns a message which you can just use, would be more readable than the current global variable/callback stuff